src/components/TodoList.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+import { TodoInfo } from './TodoInfo';
+
+type Props = {
+  todos: Todo[],
+  tempTodo: Todo | null,
+  onDelete: (id: number) => void,
+  onToggle: (id: number) => void,
+  loadingTodoId: number,
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos, tempTodo, onDelete, onToggle, loadingTodoId,
+}) => {
+  return (
+    <section className="todoapp__main">
+      {todos.map(todo => (
+        <TodoInfo
+          key={todo.id}
+          todo={todo}
+          onDelete={onDelete}
+          onToggle={onToggle}
+          isLoading={loadingTodoId === todo.id}
```

Comment:
with `loadingTodoIds` you will have te change the logic to `loadingTodoIds.includes(todo.id)`

Reply:
it doesn't work with that code because of the difference in the type (in handleDeleteTodo)